canopen/objectdictionary/__init__.py

```diff
@@ -61,10 +66,10 @@ def import_od(
     """Parse an EDS, DCF, or EPF file.
 
     :param source:
-        Path to object dictionary file or a file like object or an EPF XML tree.
-
-    :return:
-        An Object Dictionary instance.
+        Path to object dictionary file, a file like object, or an EPF XML tree.
+    :param node_id:
+        For EDS and DCF files, the node ID to use.
```

Comment:
I this param docstring should be expanded; AFAICS, the node ID is only used for calculating indexes[^1]. I'm not sure if it is a bug that it's not used for the resulting ODs `.node_id` member.

[^1]: I might be using the wrong terminology here.

Reply:
The correct terminology would be "interpolation" of values containing a `$NODEID` placeholder: https://github.com/christiansandberg/canopen/blob/5db59130423948e1e8034a67983ca00afd0269d8/canopen/objectdictionary/eds.py#L225

The `ObjectDictionary` instance is generic if loaded from an EDS file, but node-specific if loaded from a DCF (which has a `DeviceComissioning` section).  In the latter case, the `node_id` is in fact stored to the instance.  Passing the parameter is just a way to override the node ID supplied in the DCF file, and to adjust the interpolated COB-ID values in DCF- or EDS-based object dictionaries.

Reply:
> In the latter case, the `node_id` is in fact stored to the instance.

Sure, but that's the `DeviceComissioning` _Node ID_ field; not the `node_id` parameter of `import_od`.

AFAICS, the `node_id` parameter is solely used for "interpolation" via the `build_variable()` helper.

Reply:
If given, the `node_id` parameter overrides the one from the DCF: https://github.com/christiansandberg/canopen/blob/503918c9bd9ff0499a604c7805e7627ca38fb332/canopen/objectdictionary/eds.py#L90

Reply:
Sure; but `node_id` is only passed as an argument to subsequent `build_variable()` calls; it does not end up in `od.node_id`. I understood your remarks as if the latter would be true.

Reply:
Oh, you're right.  Sorry for my confusion.  Then yes, that's a bug.  When reading a DCF file and overriding the `$NODEID` interpolation value, then we shouldn't keep storing the previous ID read from the file.

The only place where the attribute is used seems to be in `BaseNode.__init__()` if no ID was passed in there as a parameter.  Seems we have too many ways of instantiating stuff...  And of course when exporting a DCF again.  In both cases, using the override from the `import_od()` argument is the sensible thing to do.

Should we fix this here or would you make a separate PR?

Reply:
For the record, the issue/PR were created:

- #478 
- #484